src/elements/image/ImageElement.tsx

```diff
@@ -83,6 +84,88 @@ export const createImageFields = (
   };
 };
 
+type ImageAsset = {
+  fields: {
+    width: number;
+    height: number;
+  };
+};
+
+const hasOwnProperty = <X extends Record<string, unknown>, Y extends string>(
+  obj: X,
+  prop: Y
+): obj is X & Record<Y, unknown> => {
+  return Object.hasOwnProperty.call(obj, prop);
+};
+
+const isRecord = (value: unknown): value is Record<string, unknown> => {
+  return typeof value === "object" && value !== null;
+};
+
+const isValidImageAsset = (
+  maybeImage: Record<string, unknown>
+): maybeImage is ImageAsset => {
+  return (
+    hasOwnProperty(maybeImage, "fields") &&
+    isRecord(maybeImage.fields) &&
+    hasOwnProperty(maybeImage.fields, "width") &&
+    hasOwnProperty(maybeImage.fields, "height") &&
+    typeof maybeImage.fields.width === "number" &&
+    typeof maybeImage.fields.height === "number"
+  );
+};
+
+const validateAssets = (maybeAssets: unknown[]) => {
```

Comment:
I've come up with a slightly cleaner way of doing this without this function but it isn't much of an improvement. Let me know if you're interested.

Reply:
Cheers for the review Sam - have implemented your other points, and definitely interested in looking into you suggestion.

Reply:
Had a good discussion about this on a call. We have come up with a solution which is cleaner and more succinct which has now been commited.